### PR TITLE
MOTECH-2034 Add a selenium test for editing a DDE

### PR DESCRIPTION
**Pull request by martokarski**

### CODE REVIEW

File: framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java [hunk 1]
```diff
@@ -16,8 +19,15 @@
     public static final By DATA_SERVICES_BUTTON = By.id("modulelink_data-services");
     public static final By SCHEMA_EDITOR_BUTTON = By.id("mdsTab_schemaEditor");
     public static final By BROWSE_INSTANCES_BUTTON = By.id("browseInstancesButton");
-    public static final By ADD_NEW_INSTANCE_BUTTON = By.id("addNewInstanceButton");
     public static final By ENTITY_SPAN = By.id("select2-chosen-2");
+    public static final By FIELD_TYPE_DROPDOWN = By.id("new-field-type");
+    public static final By FIELD_DISPLAY_NAME = By.xpath("//div[@id='new-field-displayName']/input");
+    public static final By FIELD_NAME = By.xpath("//div[@id='new-field-name']/input");
+    public static final By FIELD_TYPE_BOOLEAN = By.xpath("//div[@class='select2-result-label']/div/div/strong[text() = 'Boolean']");
```
Comment (pgesek) [inline framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java]: we should avoid to specific xpath expressions - function names or css classes can change. We can add ids to these elements in Motech if they are missing.

**Comment by martokarski** [comment]
These changes work with https://github.com/motech/motech/pull/471

**Comment by martokarski** [comment]
test this please

File: framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java [hunk 2]
```diff
@@ -42,13 +50,47 @@ public String createNewEntity(String entityName) throws InterruptedException {
...
     /**
+     * Method that goes to schema editor page and enters entity
+     * @param entityName name of the entity we want to edit
+     */
+    public void goToEditEntity(String entityName) throws InterruptedException {
+        clickWhenVisible(DATA_SERVICES_BUTTON);
+        clickWhenVisible(By.id(String.format("edit_%s", entityName)));
+    }
+
+    /**
```
Comment (kmadej) [inline framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java:69]: do not start javadoc with word 'method', in this case begin with 'Adds'.

Add fieldDisplayName field to javadocs

File: framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java [hunk 3]
```diff
@@ -42,13 +50,47 @@ public String createNewEntity(String entityName) throws InterruptedException {
...
+     * Method adds new Boolean field to the entity
+     * @param fieldName name of the field we want to add
+     */
+    public void addNewBooleanField(String fielldDisplayName, String fieldName) throws InterruptedException {
```
Comment (kmadej) [inline framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java]: misspelling, should be fieldDisplayName

File: framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java [hunk 4]
```diff
@@ -42,13 +50,47 @@ public String createNewEntity(String entityName) throws InterruptedException {
...
+     * @param entityName name of the entity we want to edit
+     */
+    public void goToEditEntity(String entityName) throws InterruptedException {
+        clickWhenVisible(DATA_SERVICES_BUTTON);
+        clickWhenVisible(By.id(String.format("edit_%s", entityName)));
+    }
+
+    /**
+     * Method adds new Boolean field to the entity
+     * @param fieldName name of the field we want to add
```
Comment (kmadej) [inline framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java]: name of the field to be added

File: framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java [hunk 5]
```diff
@@ -42,13 +50,47 @@ public String createNewEntity(String entityName) throws InterruptedException {
...
+     * Method that goes to schema editor page and enters entity
+     * @param entityName name of the entity we want to edit
+     */
+    public void goToEditEntity(String entityName) throws InterruptedException {
+        clickWhenVisible(DATA_SERVICES_BUTTON);
+        clickWhenVisible(By.id(String.format("edit_%s", entityName)));
+    }
+
+    /**
+     * Method adds new Boolean field to the entity
```
Comment (kmadej) [inline framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java]: boolean (lowercase)

File: framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java [hunk 6]
```diff
@@ -42,13 +50,47 @@ public String createNewEntity(String entityName) throws InterruptedException {
...
+        clickWhenVisible(CREATE_FIELD_BUTTON);
+        waitForElementToBeEnabled(SAVE_CHANGES_BUTTON);
+        clickWhenVisible(SAVE_CHANGES_BUTTON);
+        waitForElementToBeDisabled(SAVE_CHANGES_BUTTON);
+    }
+
+    /**
+     * Method checks if field with given name exist in table
+     * @param fieldName name of the field
+     * @return returns true if that field exists or false otherwise
```
Comment (kmadej) [inline framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java]: without word 'returns'

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

File: framework/src/main/java/org/motechproject/uitest/page/AbstractBasePage.java [hunk 7]
```diff
@@ -196,6 +196,10 @@ public void waitForElementToBeEnabled(By by) {
         waiter.until(ExpectedConditions.elementToBeClickable(by));
     }
 
+    public void waitForElementToBeDisabled(By by) {
+        waiter.until(ExpectedConditions.not(ExpectedConditions.elementToBeClickable(by)));
```
Comment (pgesek) [inline framework/src/main/java/org/motechproject/uitest/page/AbstractBasePage.java:201]: add a log statement similar to other methods

File: framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java [hunk 8]
```diff
@@ -43,14 +54,48 @@ public DataServicesPage createNewEntity(String entityName) throws InterruptedExc
     }
 
     /**
-     * Method that goes to data services page and enters entity table.
+     * Goes to schema editor page and enters entity
+     * @param entityName name of the entity we want to edit
+     */
+    public void goToEditEntity(String entityName) throws InterruptedException {
```
Comment (pgesek) [inline framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java]: can this return the DataServicesPage (this) ? I would  like to standarize on these go methods returning page objects.

Comment (pgesek) [inline framework/src/main/java/org/motechproject/uitest/page/DataServicesPage.java]: I'm also thinking about splitting this page object into 3 - one for each tab. But maybe in a separate pr

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Comment by martokarski** [comment]
test this please

**Pull request merged**